project-base/app/tests/FrontendApiBundle/Functional/Order/DeliveryAddressIsNotStoredForPacketeryOrPersonalPickupTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrontendApiBundle\Functional\Order;
+
+use App\DataFixtures\Demo\ProductDataFixture;
+use App\DataFixtures\Demo\TransportDataFixture;
+use App\Model\Product\Product;
+use App\Model\Transport\Transport;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Tests\FrontendApiBundle\Test\GraphQlWithLoginTestCase;
+
+class DeliveryAddressIsNotStoredForPacketeryOrPersonalPickupTest extends GraphQlWithLoginTestCase
+{
+    /**
+     * @param string $transportDataFixtureReference
+     * @param string $pickupPlaceIdentifier
+     */
+    #[DataProvider('getTransportDataFixtureReferences')]
+    public function testDeliveryAddressIsNotDuplicatedForLoggedInClient(
+        string $transportDataFixtureReference,
+        string $pickupPlaceIdentifier,
+    ): void {
+        $deliveryAddressesBeforeOrder = $this->getCustomersDeliveryAddresses();
+
+        $this->initializeCart($transportDataFixtureReference, $pickupPlaceIdentifier);
+
+        $orderVariables = [
+            'cartUuid' => null,
+            'city' => 'Ostrava',
+            'companyName' => 'Shopsys',
+            'companyNumber' => '12345678',
+            'companyTaxNumber' => 'CZ65432123',
+            'country' => 'CZ',
+            'deliveryAddressUuid' => null,
+            'deliveryCity' => 'Litovel',
+            'deliveryCompanyName' => null,
+            'deliveryCountry' => 'CZ',
+            'deliveryFirstName' => 'Jaromír',
+            'deliveryLastName' => 'Jágr',
+            'deliveryPostcode' => '78401',
+            'deliveryStreet' => 'Staroměstské náměstí 93',
+            'deliveryTelephone' => '605000123',
+            'email' => 'no-reply@shopsys.com',
+            'firstName' => 'Jaromír',
+            'lastName' => 'Jágr',
+            'heurekaAgreement' => true,
+            'isDeliveryAddressDifferentFromBilling' => true,
+            'newsletterSubscription' => false,
+            'note' => '',
+            'onCompanyBehalf' => true,
+            'postcode' => '70200',
+            'street' => 'Hlubinská 10',
+            'telephone' => '605000123',
+        ];
+
+        $response = $this->getResponseContentForGql(__DIR__ . '/../_graphql/mutation/CreateOrderMutation.graphql', $orderVariables);
+        $this->assertResponseContainsArrayOfDataForGraphQlType($response, 'CreateOrder');
+
+        $deliveryAddressesAfterOrder = $this->getCustomersDeliveryAddresses();
+
+        $this->assertSameSize($deliveryAddressesBeforeOrder, $deliveryAddressesAfterOrder);
+        $this->assertSame($deliveryAddressesBeforeOrder, $deliveryAddressesAfterOrder);
+    }
+
+    /**
+     * @return iterable
+     */
+    public static function getTransportDataFixtureReferences(): iterable
+    {
+        yield [TransportDataFixture::TRANSPORT_PACKETERY, '26145'];
+
+        // @todo: static uuid?
```

Comment:
what about this todo? :slightly_smiling_face: we can discuss it if you want :wink:

Reply:
🤦‍♂️ I thought about it at least three times... but still managed to move task with it

Reply:
fixed  in 5eabe1d60a